# Проект 2. Исследование надежности заемщиков


**Описание проекта** 

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.
Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

**Цель исследования** — выяснить:
1. Есть ли зависимость между количеством детей и возвратом кредита в срок?
2. Есть ли зависимость между семейным положением и возвратом кредита в срок?
3. Есть ли зависимость между уровнем дохода и возвратом кредита в срок? 
4. Как разные цели кредита влияют на его возврат в срок?

**Описание данных:**
- `children` — количество детей в семье
- `days_employed` — общий трудовой стаж в днях
- `dob_years` — возраст клиента в годах
- `education` — уровень образования клиента
- `education_id` — идентификатор уровня образования
- `family_status` — семейное положение
- `family_status_id` — идентификатор семейного положения
- `gender` — пол клиента
- `income_type` — тип занятости
- `debt` — имел ли задолженность по возврату кредитов
- `total_income` — ежемесячный доход
- `purpose` — цель получения кредита

**Ход исследования**

Данные о с информацией о клиентах получены из файла `data.csv`. О качестве данных ничего не известно. Поэтому перед проверкой зависимостей понадобится обзор данных. 

Необходимо проверить данные на ошибки и оцените их влияние на исследование. Затем, на этапе предобработки исправить самые критичные ошибки данных.
 
Таким образом, исследование пройдёт в три этапа:
 1. Обзор данных.
 2. Предобработка данных.
 3. Выявление зависимостей.

## Откроем таблицу и изучим общую информацию о данных

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('data.csv')

In [3]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

Выведем количество пропущенных значений для каждого столбца.

In [5]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце мы обработаем на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.

In [6]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. Таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработаем значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.

In [7]:
data['days_employed'] = data['days_employed'].abs()

Для каждого типа занятости выведем медианное значение трудового стажа `days_employed` в днях.

In [8]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставим их как есть. Тем более этот столбец не понадобится нам для исследования.

Выведем перечень уникальных значений столбца `children`.

In [9]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

В столбце `children` есть два аномальных значения. Удалим строки, в которых встречаются такие аномальные значения из датафрейма `data`.

In [10]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

Ещё раз выведем перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.

In [11]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков (продолжение)

Заполним пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.

In [12]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

Убедимся, что все пропуски заполнены. Ещё раз выведем количество пропущенных значений для каждого столбца с помощью двух методов.

In [13]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

Заменим вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.

In [14]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

Обработаем неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверим остальные столбцы.

In [15]:
data['education'] = data['education'].str.lower()

Выведем на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалим их.

In [16]:
data.duplicated().sum()

71

In [17]:
data = data.drop_duplicates()

### Категоризация данных

На основании диапазонов, указанных ниже, создадим в датафрейме `data` столбец `total_income_category` с категориями:

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используем собственную функцию с именем `categorize_income()` и метод `apply()`.

In [18]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [19]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

Выведем на экран перечень уникальных целей взятия кредита из столбца `purpose`.

In [20]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

Создадим функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.

Используем собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучим данные в столбце `purpose` и определим, какие подстроки помогут нам правильно определить категорию.

In [21]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [22]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

## Исследуем данные и ответим на вопросы

### Есть ли зависимость между количеством детей и возвратом кредита в срок?

Следует проверить количество клиентов в каждой из пресдтавленных групп. Возможно, что данных недостаточно для формулировки каких-либо выводов о зависимости между кол-вом детей и возвратом кредита в срок.

In [23]:
data.groupby('children')['debt'].count()

children
0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: debt, dtype: int64

Данная выборка говорит о том, что количество людей в каждой группе слишком сильно отличается. Можно утверждать, что банк имеет гораздо больше клиентов, которые совсем не имеют детей, чем остальных. Клиентов, имеющих 0 детей `14091`, а клиентов имеющих 5 детей всего `9`.

Рассмотрим самые многочисленные выборки с 0, 1 и 2-мя детьми. Создадим сводную таблицу, содержащую:
* название категории (в зависимости от количества детей)
* количество заемщиков (кредитов)
* количество должников 
* средний процент просрочки по категории 

In [24]:
data_pivot_childrens = data[data['children'] < 3].pivot_table(index='children', 
                                                              values = 'debt', 
                                                              aggfunc = ['count', 'sum', 'mean'])
data_pivot_childrens

count   sum      mean
           debt  debt      debt
children                       
0         14091  1063  0.075438
1          4808   444  0.092346
2          2052   194  0.094542

В данной таблице мы наблюдаем, что с ростом количества детей растет и процент просроченных платежей. Так среди людей, которые не имеют детей просроченных кредитов всего `7.5%` от общего числа кредитов в данной категории, у людей с одним ребенком этот параметр равняется `9.2%`, а у людей с двумя детьми - `9.5%`.

Рассмотрим полученные показатели более подробно. Разобьем категории по количеству детей на подкатегории по цели кредита и сравним средний процент просрочки в зависимости от подкатегории. Если в каждой отдельно взятой подкатегории рост невыплат сохраняется, то наши преположения верны.

In [25]:
data_pivot_childrens = data[data['children'] < 3].pivot_table(index=['children', 'purpose_category'], 
                                                              values = 'debt', 
                                                              aggfunc = ['count', 'sum', 'mean'])
data_pivot_childrens

count  sum      mean
                                   debt debt      debt
children purpose_category                             
0        операции с автомобилем    2845  243  0.085413
         операции с недвижимостью  7074  476  0.067289
         получение образования     2642  229  0.086677
         проведение свадьбы        1530  115  0.075163
1        операции с автомобилем     963  103  0.106957
         операции с недвижимостью  2447  200  0.081733
         получение образования      866   90  0.103926
         проведение свадьбы         532   51  0.095865
2        операции с автомобилем     399   48  0.120301
         операции с недвижимостью  1035   88  0.085024
         получение образования      403   46  0.114144
         проведение свадьбы         215   12  0.055814

Данная таблица содержит наглядную разницу в паказателях. С ростом количества детей растет и процент просроченных кредитов в каждой из подкатегорий. Например, возьмем цель кредита "операции с автомобилем". Клиенты без детей просрачивают в среднем не выплачивают кредит вовремя в `8.5%` случаев, у клиентов с одним ребенком этот показатель равен `10.7%`, а с двумя детьми - `12%`. 

Однако стоит заметить, что в данных имеется выброс. Для цели кредита "проведение свадьбы", среди людей, имеющийх 2 ребенка, средний процент просрочки по кредиту равен `5.6%`, что выбивается из общей логики. Возможно, выброс повился из-за того, что при наличии двух детей кредит на проведение свадьбы берут реже всего.
Данный выброс также влияет на общие показатели возврата кредита в срок в зависимости от колиичествава детей. По этой причине разница в категориях людей с одинм и двумя детьми была не такая явная на прошлом этапе аналитики. Посчитаем общие показатели, исключив подкатегорию "проведение свадьбы".

In [26]:
data_pivot_childrens = data[(data['children'] < 3) & 
                            (data['purpose_category'] != 'проведение свадьбы')].pivot_table(index='children', 
                                                                                            values = 'debt', 
                                                                                            aggfunc = ['count', 'sum', 'mean'])
data_pivot_childrens

count  sum      mean
           debt debt      debt
children                      
0         12561  948  0.075472
1          4276  393  0.091908
2          1837  182  0.099075

Исключив из анализа людей, у которых цель кредита "проведение свадьбы", мы получили средний процент невыплат по кредиту среди людей с двумя детьми равный `9.9%`, что на `0.4%` выше изначального показателя. При этом значения для клиентов без детей вообще и клиентов с один ребенком пркатически не изменились

**Вывод:** если рассмотреть самые многочисленные выборки с 0, 1 и 2-мя детьми - можно увидеть, что чем больше у заемщика детей, тем он хуже платит в срок. Клиенты, не имеющие детей, не отдают средства вовремя в среднем в `7.5%` случаев, клиенты с одним ребенком не возвращают деньги в срок в `9.2%` случаев, а с двумия детьми - в `9.9%` случаев.
Гипотеза о том, что между количеством детей и возвратом кредита в срок есть зависимость подтвердилась.

### Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [28]:
data_pivot_family_status = data.pivot_table(index='family_status', values = 'debt', aggfunc = ['count', 'sum', 'mean'])
data_pivot_family_status

count  sum      mean
                        debt debt      debt
family_status                              
Не женат / не замужем   2796  273  0.097639
в разводе               1189   84  0.070648
вдовец / вдова           951   63  0.066246
гражданский брак        4134  385  0.093130
женат / замужем        12261  927  0.075606

Можно сказать, что в разных группах средний процент просрочки по кредиту различный. Но в такимх данных сложно выделить какую-либо зависимость. Так, например, люди в статусе "Не женат / не замужем" возвращают средства не вовремя в среднем в `9.8%` процентах случаев, а в статусе "в разводе" - в `7.1%`. Разница довольно большая несмотря на то, что глобально эти две группе можно отнести к одной большой группе людей не состоящих в официальном браке. Для того, чтобы выявить более явную зависимость между симейным статусом и вовзратом кредита в срок, разделим всех клиентов на тех кто состоит в официальном браке и тех, кто не состоит. Таким образом, в первую группу попадут люди из категории "женат / замужем", это 12261 человек, а в другую группу все остальные.

In [29]:
def categorize_status(value): # категоризация клиентов по семейному статусу, value - значение family_status_id
    try:
        if value == 0: # если value = 0, то клиент относится к категории "женат / замужем"
            return 'состоит в оф. браке'
        else:
            return 'не состоит в оф. браке'
    except:
        return 'нет категории'

In [30]:
data['family_status_category'] = data['family_status_id'].apply(categorize_status)
data.head(10)

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   
5         0     926.185831         27    высшее             0   
6         0    2879.202052         43    высшее             0   
7         0     152.779569         50   среднее             1   
8         2    6929.865299         35    высшее             0   
9         0    2188.756445         41   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   
5  гражданский брак                 1      M   компаньон     0        255763   
6   женат / замужем                 0      F   компаньон     0        240525   
7   женат / замужем                 0      M   сотрудник     0        135823   
8  гражданский брак                 1      F   сотрудник     0         95856   
9   женат / замужем                 0      M   сотрудник     0        144425   

                      purpose total_income_category          purpose_category  \
0               покупка жилья                     B  операции с недвижимостью   
1     приобретение автомобиля                     C    операции с автомобилем   
2               покупка жилья                     C  операции с недвижимостью   
3  дополнительное образование                     B     получение образования   
4             сыграть свадьбу                     C        проведение свадьбы   
5               покупка жилья                     B  операции с недвижимостью   
6           операции с жильем                     B  операции с недвижимостью   
7                 образование                     C     получение образования   
8       на проведение свадьбы                     C        проведение свадьбы   
9     покупка жилья для семьи                     C  операции с недвижимостью   

   family_status_category  
0     состоит в оф. браке  
1     состоит в оф. браке  
2     состоит в оф. браке  
3     состоит в оф. браке  
4  не состоит в оф. браке  
5  не состоит в оф. браке  
6     состоит в оф. браке  
7     состоит в оф. браке  
8  не состоит в оф. браке  
9     состоит в оф. браке

В отдельном столбце `family_status_category` получили разделение всех клиентов на две категории: "состоит в оф. браке" или "не состоит в оф. браке".

In [31]:
data_pivot_family_status = data.pivot_table(index='family_status_category', values = 'debt', aggfunc = ['count', 'sum', 'mean'])
data_pivot_family_status

count  sum      mean
                         debt debt      debt
family_status_category                      
не состоит в оф. браке   9070  805  0.088754
состоит в оф. браке     12261  927  0.075606

Получаем, что клиенты, которые не состоят в официальном браке чаще запаздывают с выплатами, их показатель невозврата средств в срок равен `8.9%`, а у клиентов, которые состоят в официальном браке этот показатель равен `7.6%`.

**Вывод:** между семейным положением и возвратом кредита в срок можно пронаблюдать следующую зависимость. Состоящие в официциальном браке клиенты имеют средний процент просрочки платежей равный `7.6%`. У клиентов, которые не состоят в официальном браке, этот показатель равен `8.9%`.

### Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [32]:
len(data['total_income'].unique()) # количество уникальных значений в диапозоне зарплат

18504

Поскольку в столбце зарплат `total_income` содержит количественные переменные широкого дипозона, то будем оценивать уровень зарплат в соотвествии с ранее созданным столбцом с категориями `total_income_category`.

Построим сводную таблицу по выделенным категориям дохода.

In [33]:
data_pivot_income = data.pivot_table(index='total_income_category', values = 'debt', aggfunc = ['count', 'sum', 'mean'])
data_pivot_income

count   sum      mean
                        debt  debt      debt
total_income_category                       
A                         25     2  0.080000
B                       5014   354  0.070602
C                      15921  1353  0.084982
D                        349    21  0.060172
E                         22     2  0.090909

В категориях A, D и E выборка слишком мала для того, чтобы делать какие-либо выводы. Поскольку в категорию B попадают 5014 клиентов, а вкатегорию C попадают 15921 клиентов, то мы можем взять эти две категории для сравнения. В категории B процент клиентов, не возвративших деньги в срок, составляет `7.1%`, а в категории C - `8.5%`.

**Вывод:** в данном примере количество заемщиков во всех категориях кроме B и C несопоставимо меньше, поэтому для выявления зависимости имеет смысл сравнивать только их. Удалось у становить, что клиенты с более высоким доходом, из категории B, запаздывают с выплатами по кредиту гораздо реже, в `7.1%` случаев. Клиенты, у которых доход ниже, из категории C, запаздывают с выплатами в `8.5%` случаев. Таким образом, с ростом дохода растёт и процент кредитов, возвращенных вовремя.

### Как разные цели кредита влияют на его возврат в срок?

Проанализируем, созданный ранее столбец с категориями `purpose_category`. Построим сводную таблицу по целям кредита.

In [34]:
data_pivot_purpose = data.pivot_table(index='purpose_category', values = 'debt', aggfunc = ['count', 'sum', 'mean'])
data_pivot_purpose 

count  sum      mean
                           debt debt      debt
purpose_category                              
операции с автомобилем     4279  400  0.093480
операции с недвижимостью  10751  780  0.072551
получение образования      3988  369  0.092528
проведение свадьбы         2313  183  0.079118

Из данной таблицы видно, что доля должников менятся в зависимости от цели кредита. Так на "операции с автомобилем" приходится `9.3%` должников, что практически равно показателю в категории "получение образования" - `9.3%`, разница лишь в сотых долях процента. На "операции с недвижимостью" приходится доля должников - `7.3%`, а в категории "проведение свадьбы" процент должников составляет `7.9%`.

**Вывод:** цели кредита значительно влияют на показатель просроченных выплат по кредиту. Наибольшая доля должников приходится на цели кредита "операции с автомобилем" и "получение образования", и составляет `9.3%`. Меньше всего должников среди клиентов, которые брали кредит на "проведение свадьбы" - `7.3%` от общего числа клиентов в данной категории.

## Общий вывод.

В ходе исследования была провдеена работа с данными о платежеспособности клиентов. 

На этапе предобработки мы:
- устранили пропуски в столбцах "days_employed" и "total_income"
- удалили анамольные значения из столбца "children"
- обработали неявные дубликаты в столбце "education"
- удалили дублирующие строки
- разделили всех клиентов на категории по уровню дохода и цели кредита

Исследуя обработанные данные, удалось выяснить, что:
1. Количество детей влияет на возврат кредита в срок. Если рассмотреть самые многочисленные выборки с 0, 1 и 2-мя детьми - можно увидеть, что чем больше у заемщика детей, тем он хуже платит в срок. Клиенты, не имеющие детей, не отдают средства вовремя в среднем в `7.5%` случаев, клиенты с одним ребенком не возвращают деньги в срок в `9.2%` случаев, а с двумия детьми - в `9.9%` случаев.
2. Между семейным положением и возвратом кредита в срок есть следующая зависимость. Состоящие в официциальном браке клиенты имеют средний процент просрочки по платежам равный `7.6%`. У клиентов, которые не состоят в официальном браке, этот показатель равен `8.9%`.
3. Клиенты с более высоким доходом, из категории B (доход от 200 т.р. до 1 млн.р.), запаздывают с выплатами по кредиту гораздо реже, в `7.1%` случаев. Клиенты, у которых доход ниже, из категории C (доход от 50 т.р. до 200 т.р.), запаздывают с выплатами в `8.5%`случаев. Таким образом, чем ниже уровень дохода, тем чаще клиенты запаздывают с выплатами по кредиту.
4. Цели кредита значительно влияют на показатель просроченных выплат по кредиту. Наибольшая доля должников приходится на цели кредита "операции с автомобилем" и "получение образования", она составляет `9.3%`. Меньше всего должников среди клиентов, которые брали кредит на "проведение свадьбы" - `7.3%` от общего числа клиентов в данной категории. В категории "проведение свадьбы" процент должников составляет `7.9%`.

На этом исследование можно считать завершенным.